src/Proto.Actor/Extensions/ActorSystemExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+
+namespace Proto.Extensions
+{
+    public class ActorSystemExtensions
+    {
+        private readonly IActorSystemExtension[] _extensions = new IActorSystemExtension[100];
+        private readonly ActorSystem _actorSystem;
+
+        public ActorSystemExtensions(ActorSystem actorSystem)
+        {
+            _actorSystem = actorSystem;
+        }
+
+        public T GetExtension<T>(int extensionId) where T: IActorSystemExtension
+        {
+            return (T)_extensions[extensionId];
```

Comment:
Get an extension by type and ID. a bit ugly to require both, but perf is important